File: gdscript/src/main/java/gdscript/psi/impl/GdAssignSignImpl.java
package gdscript.psi.impl;

import org.jetbrains.annotations.*;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import com.intellij.extapi.psi.ASTWrapperPsiElement;
import gdscript.psi.*;

public class GdAssignSignImpl extends ASTWrapperPsiElement implements GdAssignSign {

  public GdAssignSignImpl(@NotNull ASTNode node) {
    super(node);
  }

  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitAssignSign(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

}

File: gdscript/src/main/java/gdscript/psi/impl/GdAssignSignImpl.kt
package gdscript.psi.impl

import com.intellij.extapi.psi.ASTWrapperPsiElement
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import gdscript.psi.GdAssignSign
import gdscript.psi.GdVisitor

class GdAssignSignImpl(node: ASTNode) : ASTWrapperPsiElement(node), GdAssignSign {
    fun accept(visitor: GdVisitor) {
        visitor.visitAssignSign(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }
}
